### DISCUSSION TITLE
Applying type-graphql would reduce pain from working with GraphQL

### DISCUSSION THREAD

**Discussion by darron1217** [strapi/strapi#7952] [Legacy Topics]
https://github.com/MichalLytek/type-graphql

Recently I found this framework.

If there's a plan to add Typescript support to Strapi, it would be nice to have `type-graphql` in our GraphQL plugin.

This will make it easier for Strapi developers to work with GraphQL.

Anyway, thanks for the great work :)

**Comment by derrickmehaffy** [comment]
Thank you for the feedback, for the moment we don't have any concrete plans for TS yet (a handful of PRs, ect) as it would require a lot of reworking of the core framework, maybe sometime in 2021

**Reply by darron1217** [reply]
Got it Thanks 👍
